WordPress/Classes/Stores/StatsInsightsCache.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+final class StatsInsightsCache {
+    static let shared = StatsInsightsCache()
```

Comment:
I'm not sure if the cache needs to be a singleton, given its users the "Store" types are already singletons.

Reply:
The same cache is used between `StatsRevampStore` and `StatsPeriodStore`: they access the same data, so I think it's a good idea to make it a singleton to ensure the data isn't lost.

`StatsInsightsCache` is currently only used in one store, but I made it a singleton for consistency. Plus, making it a singleton is OK because you shouldn't be creating more than one instance, or it loses its purpose. The way the stores are created, on the hand, could change.